### DISCUSSION TITLE
Having Trouble Getting Breadcrumbs Working Using `useRouterState`

### DISCUSSION THREAD

**Discussion by harrismcc** [TanStack/router#2332] [Q&A]
I'm trying to build a breadcrumbs component that can automatically use the router state to show the correct breadcrumbs for a page. However, I'm having trouble.

I followed the documentation for breadcrumbs and am getting the router state like `const matches = useRouterState({ select: (s) => s.matches });` but that gives me an array with the routes `['__root__', '/animals/test']` where ideally I'd have `['animals', 'animals/test']`.

My File structure looks like:
![CleanShot 2024-09-13 at 13 50 24](https://github.com/user-attachments/assets/79838682-d089-4480-9da7-a049982211a6)

What am I missing here? Seems like the route-matching sees `animals/test` as totally separate than `/animals`, but I want there to be a parent-child relationship (even though they aren't nested components, and only share the common layout for the whole app`.

Any ideas here for what I'm doing wrong? Thanks!

**Comment by SeanCassiere** [comment]
In this example structure posted, `/animals` wouldn't be matched when on the `/animals/test` route. This is because the `animals/index` route is separate from `animals/test`. You could have a `/routes/animals.tsx` file without any components which'd then always match when you are on a child of the `animals` path.
